cli/cmd/inject.go

```diff
@@ -503,3 +521,23 @@ func overwriteRegistry(image, newRegistry string) string {
 	}
 	return registry + imageName
 }
+
+func shouldInjectThrowError(report *inject.Report) (bool, error) {
+	if !report.AutomountServiceAccountToken {
+		return true, errors.New(automountServiceAccountTokenError)
+	}
+
+	if report.HostNetwork {
+		return true, errors.New(hostNetworkError)
+	}
+
+	if report.Sidecar {
+		return true, errors.New(sidecarError)
+	}
+
+	if report.UDP {
+		return true, errors.New(udpError)
+	}
+
+	return false, nil
+}
```

Comment:
I think this can be simplified by just returning `error`; nil if not, `error` if yes

Reply:
Even better, a `[]error` would be cooler in order to accumulate all the errors from a single pod instead of short-circuiting on the first :wink: